util/mcf_dict_util.py

```diff
@@ -0,0 +1,537 @@
+
+    Args:
+        new_path: Glob like path to open a set of files and create a list of node for each file.
+        existing_dict: (Optional) dict returned from previous call to add more files to same dict object.
+        reopen: Boolean value to skip opening files already present in the dictionary to prevent overwrites.
+        
+    Returns:
+        A dictionary with filenames as objects and a list of OrderedDict objects where each object 
+            represents a node in the MCF file as value.
+    """
+    if not existing_dict:
+        existing_dict = {}
+
+    new_path = os.path.expanduser(new_path)
+    # if is dir add **/*.mcf
+    if os.path.isdir(new_path):
+        new_path = os.path.join(new_path, '**/*.mcf')
+
+    matching_files = glob.glob(new_path, recursive=True)
+    for cur_file in matching_files:
+        # set of files
+        print(cur_file)
+        if cur_file not in existing_dict or reopen:
+            existing_dict[cur_file] = mcf_file_to_dict_list(cur_file)
+
+    return existing_dict
+
+
+def dict_list_to_mcf_str(node_list: list,
+                         sort_keys: bool = False,
+                         regen_complex_vals: bool = False) -> str:
+    """Creates a string (which could be written to a file) from list of Dict like object representation of nodes.
+
+    Args:
+        node_list: List of OrderedDict or Dict objects representing a node in MCF file.
+        sort_keys: Boolean value to sort the properties before appending the node to the string.
+                    NOTE: This keeps Node on the top and all inline comments would be moved around.
+        regen_complex_vals: Boolean value to regenrate the value string of the property from the list 
+                                of complex value representation.
+        
+    Returns:
+        String representation of the list of nodes.
+    """
+    ret_str = ''
+    for _node in node_list:
+        cur_node = _node.copy()
+
+        is_comment_block = True
+        for prop in cur_node:
+            if not prop.startswith('__comment'):
+                is_comment_block = False
+        # TODO other validations if required
+        if not is_comment_block:
+            if 'Node' not in cur_node:
+                raise ValueError('Each node must have Node: <name>".')
+
+        # preserve comments
+        if '__comment' in cur_node:
+            ret_str += cur_node['__comment']
+        cur_node.pop('__comment', None)
+        # add comments till node
+        prop_list = list(cur_node.keys())
+        i = 0
+        while i < len(prop_list) and prop_list[i].startswith('__comment'):
+            ret_str += cur_node[prop_list[i]]
+            ret_str += '\n'
+            cur_node.pop(prop_list[i], None)
+            i += 1
+
+        if not is_comment_block:
+            # Keep Node: first
+            ret_str += f"{'Node'}: {cur_node['Node']['namespace']+':' if cur_node['Node']['namespace'] else ''}{cur_node['Node']['value']}"
```

Comment:
Could add a check for `if 'Node' in cur_node:`

Reply:
skipped the check for comment blocks which would fail the condition and hence be dropped in the output.